warehouse/manage/forms.py

```diff
@@ -61,10 +61,20 @@ class AddEmailForm(NewEmailMixin, forms.Form):
 
     __params__ = ["email"]
 
-    def __init__(self, *args, user_service, **kwargs):
+    def __init__(self, *args, request, user_service, **kwargs):
```

Comment:
We don't need to pass the entire `request` in to the form. Instead we can just pass in `request.user.id` as `user_id`, and have it default to `None` if it's not provided.